commands/purger.js

```diff
@@ -0,0 +1,118 @@
+const { SlashCommandBuilder } = require('@discordjs/builders')
+const { foemp } = require('../helpers/foemp')
+const { reply, defer } = require('../helpers/interactionHelper')
+const { verifyAdminAsync } = require('./admin')
+const { getGuild, saveGuild } = require('../helpers/guildData')
+const { addPurgerAndStartTask, removePurgeChannelTask } = require('../tasks/purgeChannel')
+const { MessageEmbed } = require('discord.js')
+
+const cronSyntaxRegex = /(((\d+,)+\d+|(\d+(\/|-)\d+)|\d+|\*) ?){5}/
+
+const addNewPurger = async interaction => {
+  if (!await verifyAdminAsync(interaction)) { return }
+
+  const description = interaction.options.getString('description')
+  const cronTime = interaction.options.getString('cron_time') ?? '0 0 * * *'
+  const maxAge = interaction.options.getInteger('max_age')
+  const channel = interaction.channel
+  const guild = await getGuild(interaction.guildId)
+
+  if (channel.isVoiceBased) {
```

Comment:
The bot should also check whether it is allowed to remove messages in the requested channel